blueocean-github-pipeline/src/test/java/io/jenkins/blueocean/blueocean_github_pipeline/GithubOrgFolderPermissionsTest.java

```diff
@@ -0,0 +1,166 @@
+
+    @Test
+    public void canCreateWhenHavePermissionsOnDefaultOrg() throws Exception {
+        MockAuthorizationStrategy authz = new MockAuthorizationStrategy();
+        authz.grant(Jenkins.ADMINISTER).everywhere().to(user);
+        j.jenkins.setAuthorizationStrategy(authz);
+        // refresh the JWT token otherwise all hell breaks loose.
+        jwtToken = getJwtToken(j.jenkins, "vivek", "vivek");
+        createGithubOrgFolder(true);
+    }
+
+    @Test
+    public void canNotCreateWhenHaveNoPermissionOnDefaultOrg() throws Exception {        
+        MockAuthorizationStrategy authz = new MockAuthorizationStrategy();
+        authz.grant(Item.READ).everywhere().to(user);
+        j.jenkins.setAuthorizationStrategy(authz);
+        // refresh the JWT token otherwise all hell breaks loose.
+        jwtToken = getJwtToken(j.jenkins, "vivek", "vivek");
+        createGithubOrgFolder(false);
+    }
+
+    @Test
+    public void canCreateWhenHavePermissionsOnCustomOrg() throws Exception {
+        MockAuthorizationStrategy authz = new MockAuthorizationStrategy();
+        authz.grant(Item.READ).everywhere().to(user);
+        authz.grant(Item.CREATE, Item.CONFIGURE).onFolders(getOrgRoot()).to(user);
+        j.jenkins.setAuthorizationStrategy(authz);
+        // refresh the JWT token otherwise all hell breaks loose.
+        jwtToken = getJwtToken(j.jenkins, "vivek", "vivek");
+        createGithubOrgFolder(true);
+    }
+
+    @Test
+    public void canNotCreateWhenHaveNoPermissionOnCustomOrg() throws Exception {
+        MockAuthorizationStrategy authz = new MockAuthorizationStrategy();
+        authz.grant(Item.READ).everywhere().to(user);
+        j.jenkins.setAuthorizationStrategy(authz);
+        // refresh the JWT token otherwise all hell breaks loose.
+        jwtToken = getJwtToken(j.jenkins, "vivek", "vivek");
+        createGithubOrgFolder(false);
+    }
+
+    private void createGithubOrgFolder(boolean shouldSuceed) throws Exception {
+        String credentialId = createGithubCredential(user);
+        String orgFolderName = "cloudbeers1";
+        Map resp = new RequestBuilder(baseUrl)
+                .status(shouldSuceed ? 201 : 403)
+                .jwtToken(getJwtToken(j.jenkins,user.getId(), user.getId()))
+                .post("/organizations/" + getOrgName() + "/pipelines/")
+                .data(ImmutableMap.of("name", orgFolderName,
+                        "$class", "io.jenkins.blueocean.blueocean_github_pipeline.GithubPipelineCreateRequest",
+                        "scmConfig", ImmutableMap.of("config",
+                                ImmutableMap.of("repos", ImmutableList.of("PR-demo"), "orgName","cloudbeers"),
+                                "credentialId", credentialId,
+                                "uri", githubApiUrl)
+                ))
+                .build(Map.class);
+
+        TopLevelItem item = getOrgRoot().getItem(orgFolderName);
```

Comment:
the creation is (should be) synchronous  (hence a 201) - code was copied from other tests